mulog-kinesis/src/com/brunobonacci/mulog/publishers/kinesis.clj

```diff
@@ -0,0 +1,71 @@
+(ns com.brunobonacci.mulog.publishers.kinesis
+  (:require [com.brunobonacci.mulog.buffer :as rb]
+            [com.brunobonacci.mulog.utils :as ut]
+            [com.brunobonacci.mulog.publishers.aws-utils :as awsutils]
+            [cheshire.core :as json]
+            [cheshire.generate :as gen]))
+
+;;
+;; Add Flake encoder to JSON generator
+;;
+(gen/add-encoder com.brunobonacci.mulog.core.Flake
+                 (fn [x ^com.fasterxml.jackson.core.JsonGenerator json]
+                   (gen/write-string json ^String (str x))))
+
+(defn- put-records
+  [{:keys [kinesis-client-params stream-name partition-key-name format] :as config} records]
+  (let [kinesis-client (awsutils/create-kinesis-client kinesis-client-params)
```

Comment:
here you are creating a client on every put records.
 i think it is better to create the client in `kinesis-publisher` and pass it to the constructor

Reply:
agree